src/proc/constants.rs

```diff
@@ -142,7 +143,33 @@ impl<'a> ConstantSolver<'a> {
                         Ok(components[index])
                     }
                 }
-                crate::TypeInner::Matrix { .. } => todo!(),
+                crate::TypeInner::Matrix {
```

Comment:
why is that `match self.types[ty].inner` needed at all? If the constant is `Composite`, and you need to access the `index`, you should be able to just get `components[index]` and be done with it, regardless of the type.

Reply:
This match serves 3 purposes:
1. A sanity check that the type can be accessed
2. Allows us to extract information to bound check so that we don't panic (the `ConstantSolver` doesn't rely on validation having ran, I can change this if you want)
3. Most importantly, matrix access indexes an entire slice and not just one element